includes/class-wc-fgc-update-variation-cart.php

```diff
@@ -0,0 +1,387 @@
+<?php
+/**
+ * Update Variations in cart.
+ *
+ * Handles all Variation Update related functionalities.
+ *
+ *
+ * @since 1.0.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+if ( class_exists( 'WC_FGC_Update_Variation_Cart' ) ) {
+	return;
+}
+
+/**
+ * Main WC_FGC_Update_Variation_Cart Class
+ *
+ * @class WC_FGC_Update_Variation_Cart
+ * @package Class
+ * @author   Kevin
+ * @version 1.0.0
+ */
+class WC_FGC_Update_Variation_Cart {
+	/**
+	 * The plugin version
+	 *
+	 * @var string
+	 */
+	public static $version = '1.0.0';
+	/**
+	 * The plugin name
+	 *
+	 * @var string
+	 */
+	public static $name = 'wc_fgc';
+
+	/**
+	 * WC Update Variation Cart constructor
+	 *
+	 * @since 1.0.0
+	 */
+	public static function init() {
+
+		// Enqueue required js and css.
+		add_filter( 'wp_enqueue_scripts', array( __CLASS__, 'enqueue_cart_script' ) );
+
+		// Add edit option on the cart page.
+		add_action( 'woocommerce_after_cart_item_name', array( __CLASS__, 'variation_update_data' ), 1, 2 );
+
+		// Handle the ajax request of the update cart.
+		add_action( 'wp_ajax_wc_fgc_get_product_html', array( __CLASS__, 'get_variation_html' ) );
+		add_action( 'wp_ajax_nopriv_wc_fgc_get_product_html', array( __CLASS__, 'get_variation_html' ) );
+
+		// Update the cart as per the user choice.
+		add_action( 'wp_ajax_wc_fgc_update_variation_in_cart', array( __CLASS__, 'update_variation_in_cart' ) );
+		add_action( 'wp_ajax_nopriv_wc_fgc_update_variation_in_cart', array( __CLASS__, 'update_variation_in_cart' ) );
+
+		// Show error message on the checkout page.
+		add_action( 'woocommerce_before_checkout_process', array( __CLASS__, 'check_is_valid_product' ) );
+	}
+
+	/**
+	 * Enqueue needed js and css
+	 *
+	 * @name enqueue_cart_script
+	 * @param string $page   Name of the page.
+	 * @since 1.0.0
+	 */
+	public static function enqueue_cart_script( $page ) {
+
+		// check is cart page or not.
+		if ( is_cart() ) {
+			// Hook
+			add_filter( 'woocommerce_get_script_data', array( __CLASS__, 'change_variation_select_alert' ), 10, 2 );
+
+			$min_ext = ( ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ? '' : 'min' );
+
+			// Enqueue default js for woocommerce.
+			wp_enqueue_script( 'wc-add-to-cart-variation' );
+
+			//Alter the add to cart variation text
+			// Enqueue js.
+			wp_enqueue_script( self::$name . '_js', plugin_dir_url( _WC_FGC_PLUGIN_NAME ) . 'assets/js/frontend-variation-cart' . $min_ext . '.js', array( 'jquery', 'flexslider', 'wc-single-product' ), self::$version, false );
```

Comment:
any reason we can't move the script to the footer?

Reply:
no actually, just left that as it was.

Reply:
my fault then! past!kathy dropped the ball

Reply:
pushed it to the bottom though